### PR TITLE
More refactoring for plugin manager.

### PR DESCRIPTION
**Pull request by jezdez**
### Description

Following https://github.com/conda/conda/pull/13057/files#r1328506001 we noticed that #13057 removed the duplication check and makes the plugin loading prone to import time side effects.

### Checklist - did you ...

- ~[ ] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?~
     
- ~[ ] Add / update necessary tests?~
- ~[ ] Add / update outdated documentation?~

### CODE REVIEW

**Review by kenodegard — CHANGES_REQUESTED** [review]
After reviewing `pluggy.PluginManager.register` again the only instances where a `ValueError` is raised is when the plugin has already been registered. Let's just silently consume this instead of converting it into a PluginError.

File: conda/plugins/manager.py [hunk 1]
```diff
@@ -74,6 +74,24 @@ def get_canonical_name(self, plugin: object) -> str:
...
+        """
+        try:
+            # register plugin but ignore ValueError since that means
+            # the plugin has already been registered
+            return super().register(plugin, name=name)
+        except ValueError:
+            return None
+        except Exception as err:
+            raise PluginError(
+                f"Error while loading first-party conda plugin: "
```
Comment (kenodegard) [inline conda/plugins/manager.py:91]:
```suggestion
                f"Error while loading conda plugin: "
```

File: conda/plugins/manager.py [hunk 2]
```diff
@@ -74,6 +74,24 @@ def get_canonical_name(self, plugin: object) -> str:
...
+                f"{self.get_canonical_name(plugin)} ({err})"
```
Comment (kenodegard) [inline conda/plugins/manager.py:92]:
```suggestion
                f"{name or self.get_canonical_name(plugin)} ({err})"
```

File: tests/plugins/test_subcommands.py [hunk 3]
```diff
@@ -76,14 +75,12 @@ def test_duplicated(plugin_manager, conda_cli: CondaCLIFixture):
...
-        plugin_manager.load_plugins(plugin)
+    # invalid, identical plugins, error ignored
+    plugin_manager.load_plugins(plugin)
 
-    # invalid, similar plugins
+    # invalid, similar plugins, error ignored
     plugin2 = SubcommandPlugin(name="custom", summary="Summary.")
-    with pytest.raises(PluginError, match="Error while loading first-party"):
-        plugin_manager.load_plugins(plugin2)
+    plugin_manager.load_plugins(plugin2)
```
Comment (kenodegard) [inline tests/plugins/test_subcommands.py:83]:
```suggestion
    assert plugin_manager.load_plugins(plugin2) == 0
```

File: tests/plugins/test_subcommands.py [hunk 4]
```diff
@@ -76,14 +75,12 @@ def test_duplicated(plugin_manager, conda_cli: CondaCLIFixture):
     plugin = SubcommandPlugin(name="custom", summary="Summary.")
     assert plugin_manager.load_plugins(plugin) == 1
 
-    # invalid, identical plugins
-    with pytest.raises(PluginError, match="Error while loading first-party"):
-        plugin_manager.load_plugins(plugin)
+    # invalid, identical plugins, error ignored
+    plugin_manager.load_plugins(plugin)
```
Comment (kenodegard) [inline tests/plugins/test_subcommands.py:79]:
```suggestion
    assert plugin_manager.load_plugins(plugin) == 0
```

**Comment by jezdez** [comment]
Doc build error:

```
Running Sphinx v4.2.0

Traceback (most recent call last):
  File "/home/docs/checkouts/readthedocs.org/user_builds/continuumio-conda/envs/13118/lib/python3.10/site-packages/sphinx/config.py", line 328, in eval_config_file
    exec(code, namespace)
  File "/home/docs/checkouts/readthedocs.org/user_builds/continuumio-conda/checkouts/13118/docs/source/conf.py", line 14, in <module>
    import conda
  File "/home/docs/checkouts/readthedocs.org/user_builds/continuumio-conda/checkouts/13118/conda/__init__.py", line 9, in <module>
    from .__version__ import __version__
  File "/home/docs/checkouts/readthedocs.org/user_builds/continuumio-conda/checkouts/13118/conda/__version__.py", line 4
    __version__ : str = version : str = '23.7.5.dev89'
                                ^
SyntaxError: invalid syntax

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/docs/checkouts/readthedocs.org/user_builds/continuumio-conda/envs/13118/lib/python3.10/site-packages/sphinx/cmd/build.py", line 276, in build_main
    app = Sphinx(args.sourcedir, args.confdir, args.outputdir,
  File "/home/docs/checkouts/readthedocs.org/user_builds/continuumio-conda/envs/13118/lib/python3.10/site-packages/sphinx/application.py", line 216, in __init__
    self.config = Config.read(self.confdir, confoverrides or {}, self.tags)
  File "/home/docs/checkouts/readthedocs.org/user_builds/continuumio-conda/envs/13118/lib/python3.10/site-packages/sphinx/config.py", line 172, in read
    namespace = eval_config_file(filename, tags)
  File "/home/docs/checkouts/readthedocs.org/user_builds/continuumio-conda/envs/13118/lib/python3.10/site-packages/sphinx/config.py", line 331, in eval_config_file
    raise ConfigError(msg % err) from err
sphinx.errors.ConfigError: There is a syntax error in your configuration file: invalid syntax (__version__.py, line 4)


Configuration error:
There is a syntax error in your configuration file: invalid syntax (__version__.py, line 4)

```

CleanShot 2023-09-20 at 15 16 41@2x

**Review by kenodegard — APPROVED** [review]

**Pull request merged**